h/static/scripts/directive/excerpt.coffee

```diff
@@ -0,0 +1,48 @@
+###*
+# @ngdoc directive
+# @name excerpt
+# @restrict AE
+# @description Checks to see if text is overflowing its container.
+# If so, it prepends/appends expanders/collapsers. Note, to work
+# the element needs a max height.
+###
+module.exports = ->
+  link: (scope, elem, attr, ctrl) ->
+    scope.collapsed = true
+
+    scope.isOverflowing = ->
+      excerpt = elem[0].querySelector('.excerpt')
+      if excerpt == null
+        return false
+      return elem[0].scrollHeight > elem[0].clientHeight
+
+    scope.toggle = ->
+      scope.collapsed = !scope.collapsed
+
+  scope:
+    enabled: '=excerptIf'
+    maxheight: '=maxheight'
+    bottomGradient: '=excerptBottomGradient'
+  restrict: 'AE'
+  transclude: true
+  template: '''
+    <ng-transclude></ng-transclude>
```

Comment:
Not sure how this isn't breaking everything, but this should be conditional on `enabled`. i.e. `<ng-transclude ng-if="!enabled"></ng-transclude>`